pod/annotations.go

```diff
@@ -626,3 +626,54 @@ func SidecarAnnotations(sidecarName, suffix string) []string {
 		LegacySidecarAnnotation(sidecarName, suffix),
 	}
 }
+
+type PlatformSidecar struct {
+	Name     string
+	Channel  string
+	ArgsJSON []byte
+}
+
+// PlatformSidecars parses sidecar-related annotations and returns a structured
+// slice of platform sidecars.
+func PlatformSidecars(annotations map[string]string) []PlatformSidecar {
+	// Look for legacy include annotation
+	var sidecarNames []string
+	if sidecarList := annotations[AnnotationKeySidecarsIncludeLegacy]; sidecarList != "" {
+		sidecarNames = strings.Split(sidecarList, " ")
+	}
+
+	// Handle the current include annotation format
+	for a, val := range annotations {
+		if strings.HasSuffix(a, "."+AnnotationKeySuffixSidecars) {
+			if boolVal, _ := strconv.ParseBool(val); boolVal {
+				sidecarNames = append(sidecarNames, strings.TrimSuffix(a, "."+AnnotationKeySuffixSidecars))
+			}
+		}
+	}
+
+	var sidecars []PlatformSidecar
+	for _, sidecarName := range sidecarNames {
+		sidecar := PlatformSidecar{
+			Name:    sidecarName,
+			Channel: "default",
+		}
+		for _, key := range SidecarAnnotations(sidecarName, "channel") {
+			if channel, ok := annotations[key]; ok {
+				sidecar.Channel = channel
+				break
+			}
+		}
+
+		for _, key := range []string{
+			SidecarAnnotation(sidecarName, "arguments"),
+			LegacySidecarAnnotation(sidecarName, "args"),
```

Comment:
Do we want to make `SidecarAnnotation`, `LegacySidecarAnnotation`, and `SidecarAnnotations` unexported?

Reply:
`SidecarAnnotation` is still used by the mutator to add `channel-override` and `channel-override-reason` annotations to the pod.

For consistency, I suggest we leave them all exported. The other two will be removed soon.